scipy/ndimage/src/_ni_measure.pyx

```diff
@@ -0,0 +1,173 @@
+
+cimport cython
+from cython cimport sizeof
+import numpy as np
+cimport numpy as np
+
+np.import_array()
+
+cdef extern from *:
+   ctypedef int Py_intptr_t
+
+ctypedef void (*PyArray_CopySwapFunc)(void *, void *, int, void *)
+
+cdef extern from "numpy/arrayobject.h" nogil:
+    ctypedef struct PyArrayIterObject:
+        np.intp_t *coordinates
+        char *dataptr
+        np.npy_bool contiguous
+
+    ctypedef struct PyArrayObject:
+        int nd
+
+    void *PyDataMem_NEW(size_t)
+    void PyDataMem_FREE(void *)
+
+
+######################################################################
+# Use of Cython's type templates for type specialization
+######################################################################
+
+# Only integer values are allowed.
+ctypedef fused data_t:
+    np.int8_t
+    np.int16_t
+    np.int32_t
+    np.int64_t
+    np.uint8_t
+    np.uint16_t
+    np.uint32_t
+    np.uint64_t
+
+#####################################################################
+# Function Specializers and asociate function for using fused type
+#####################################################################
+
+ctypedef void (*func_p)(void *data, np.flatiter iti, np.intp_t max_label, 
+               int *regions, int rank) nogil
+
+def get_funcs(np.ndarray[data_t] input):
+    return (<Py_intptr_t> findObjectsPoint[data_t])
+
+######################################################################
+# Update Regions According to Input Data Type
+######################################################################
+
+cdef inline findObjectsPoint(data_t *data, np.flatiter _iti, np.intp_t max_label, 
+            int *regions, int rank):
+    cdef int kk =0
+    cdef np.intp_t cc
+
+    # only integer or boolean values are allowed, since s_index is being used in indexing
+    cdef np.intp_t s_index = <np.intp_t> (<data_t *> data)[0] - 1
+
+    if s_index >=0  and s_index < max_label:
+        if rank > 0:
+            s_index *= 2 * rank
+            if regions[s_index] < 0:
+                for kk in range(rank):
+                    cc = (<PyArrayIterObject *>_iti).coordinates[kk]
+                    regions[s_index + kk] = cc
+                    regions[s_index + kk + rank] = cc + 1
+
+            else:
+                for kk in range(rank):
+                    cc = (<PyArrayIterObject *>_iti).coordinates[kk]
+                    if cc < regions[s_index + kk]:
+                        regions[s_index + kk] = cc
+                    if cc + 1 > regions[s_index + kk + rank]:
+                        regions[s_index + kk + rank] = cc + 1
+        else:
+            regions[s_index] = 1
+
+    return 1
+
+######################################################################
+# Implementaion of find_Objects function:-
+######################################################################
+
+cpdef _findObjects(np.ndarray input_raw, np.intp_t max_label):
+    cdef:
+        funcs = get_funcs(input_raw.take([0]))
```

Comment:
Why not:

```
cpdef _findObjects(np.ndarray[data_t] input_raw, np.intp_t max_label):
    cdef
        func_p findObjectsPoint = funcs[data_t]
```

If this works, you can get rid of the `get_funcs` function altogether.